ui/src/app/applications/components/applications-list/applications-tiles.tsx

```diff
@@ -100,156 +101,160 @@ export const ApplicationTiles = ({applications, syncApplication, refreshApplicat
-                                                <span className='applications-list__title'>{app.metadata.name}</span>
-                                            </div>
-                                        </div>
-                                        <div className='row'>
-                                            <div className='columns small-3' title='Project:'>
-                                                Project:
-                                            </div>
-                                            <div className='columns small-9'>{app.spec.project}</div>
-                                        </div>
-                                        <div className='row'>
-                                            <div className='columns small-3' title='Labels:'>
-                                                Labels:
-                                            </div>
-                                            <div className='columns small-9'>
-                                                <Tooltip
-                                                    zIndex={4}
-                                                    content={
-                                                        <div>
-                                                            {Object.keys(app.metadata.labels || {})
-                                                                .map(label => ({label, value: app.metadata.labels[label]}))
-                                                                .map(item => (
-                                                                    <div key={item.label}>
-                                                                        {item.label}={item.value}
-                                                                    </div>
-                                                                ))}
+                <DataLoader load={() => services.viewPreferences.getPreferences()}>
+                    {pref => (
+                        <div className='applications-tiles argo-table-list argo-table-list--clickable row small-up-1 medium-up-2 large-up-3 xxxlarge-up-4' ref={appContainerRef}>
+                            {applications.map((app, i) => (
+                                <div key={app.metadata.name} className='column column-block'>
+                                    <div
+                                        ref={appRef.set ? null : appRef.ref}
+                                        className={`argo-table-list__row applications-list__entry applications-list__entry--health-${app.status.health.status} ${
+                                            selectedApp === i ? 'applications-tiles__selected' : ''
+                                        }`}>
+                                        <div className='row' onClick={e => ctx.navigation.goto(`/applications/${app.metadata.name}`, {view: pref.appDetails.view}, {event: e})}>
```

Comment:
The application details page already loads view preferences and uses the correct view type. As far as I understand there is no specify view type in query parameter here.

Reply:
Hi @alexmt, yeah, it loads and selects the proper view fine, initially.  It's just that we need to properly register it in the navigation history. Try this, with and without the changes to this file, for comparison:

1. Launch Argo CD UI tab, with existing apps already created
2. Drill down to the details view of an application
3. Switch to another view of the app (eg network or pods)
4. Press the Back button
--> It does not change the view

Reply:
Actually. we need to force using the query parameter, otherwise, the view doesn't change because the current view is actually the preferred view.